backend/infrahub/graphql/manager.py

```diff
@@ -101,70 +101,75 @@ def get_attr_kind(node_schema: MainSchemaTypes, attr_schema: AttributeSchema) ->
 
 @dataclass
 class BranchDetails:
-    branch_name: str
     schema_changed_at: Timestamp
     schema_hash: str
     gql_manager: GraphQLSchemaManager
 
 
 class GraphQLSchemaManager:
     _extra_types: dict[str, GraphQLTypes] = {}
-    _branch_details_by_name: dict[str, BranchDetails] = {}
+    _branch_details_by_hash: dict[str, BranchDetails] = {}
+    _branch_name_by_hash: dict[str, set[str]] = {}
+    _branch_hash_activation_by_branch_name: dict[str, dict[str, str]] = {}
 
     @classmethod
     def clear_cache(cls) -> None:
-        cls._branch_details_by_name = {}
+        cls._branch_details_by_hash = {}
+        cls._branch_name_by_hash = {}
 
     @classmethod
     def purge_inactive(cls, active_branches: list[str]) -> set[str]:
```

Comment:
Are we automatically purging the old schema when a branch with a different schema is being deleted ?